kernel/arch/dreamcast/sound/snd_stream.c

```diff
@@ -503,124 +552,153 @@ void snd_stream_stop(snd_stream_hnd_t hnd) {
     chan->cmd = AICA_CH_CMD_STOP;
     snd_sh4_to_aica(tmp, cmd->size);
 
-    /* Channel 1 */
-    cmd->cmd_id = streams[hnd].ch[1];
-    snd_sh4_to_aica(tmp, cmd->size);
+    if(streams[hnd].channels == 2) {
+        /* Channel 1 */
+        cmd->cmd_id = streams[hnd].ch[1];
+        snd_sh4_to_aica(tmp, cmd->size);
+    }
 }
 
 /* The DMA will chain to this to start the second DMA. */
 static uint32 dmadest, dmacnt;
+static void dma_done(ptr_t data) {
```

Comment:
```suggestion
static inline void dma_done(ptr_t data) {
```

Reply:
I thought we were doing away with ptr_t?

```suggestion
static inline void dma_done(void *data) {
```